spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplateTests.java

```diff
@@ -0,0 +1,261 @@
+		eout = bindings.get(0);
+		assertEquals(Binding.DestinationType.EXCHANGE, eout.getDestinationType());
+		assertEquals(exchange1.getName(), eout.getExchange());
+		assertEquals(exchange2.getName(), eout.getDestination());
+
+		admin.deleteExchange(exchange1.getName());
+		admin.deleteExchange(exchange2.getName());
+	}
+
+	@Test
+	public void testSpecificExchange() {
+		RabbitAdmin admin = new RabbitAdmin(connectionFactory);
+		Map<String, Object> args = Collections.<String, Object>singletonMap("alternate-exchange", "");
+		Exchange exchange = new DirectExchange(UUID.randomUUID().toString(), true, true, args);
+		admin.declareExchange(exchange);
+		Exchange exchangeOut = this.template.exchange("/", exchange.getName());
+		assertTrue(exchangeOut.isDurable());
+		assertTrue(exchangeOut.isAutoDelete());
+		assertEquals(exchange.getName(), exchangeOut.getName());
+		assertEquals(args, exchangeOut.getArguments());
+		admin.deleteExchange(exchange.getName());
+	}
+
+	@Test
+	public void testSpecificQueue() {
+		RabbitAdmin admin = new RabbitAdmin(connectionFactory);
+		Map<String, Object> args = Collections.<String, Object>singletonMap("foo", "bar");
+		Queue queue = new Queue(UUID.randomUUID().toString(), true, false, true, args);
+		admin.declareQueue(queue);
+		Queue queueOut = this.template.queue("/", queue.getName());
+		assertTrue(queueOut.isDurable());
+		assertTrue(queueOut.isAutoDelete());
+		assertEquals(queue.getName(), queueOut.getName());
+		assertEquals(args, queueOut.getArguments());
+		admin.deleteQueue(queue.getName());
+	}
+
+	@Test
+	public void testGenericGet() {
+		RabbitAdmin admin = new RabbitAdmin(connectionFactory);
+		Exchange exchange = new DirectExchange(UUID.randomUUID().toString(), false, true);
+		admin.declareExchange(exchange);
+		@SuppressWarnings("unchecked")
+		List<Map<String, Object>> connections = this.template.executeGet(List.class, new String[] { "connections" });
+		assertTrue(connections.size() > 0);
+		admin.deleteExchange(exchange.getName());
+	}
+
+	@Test
+	public void testPolicies() {
+		String name = UUID.randomUUID().toString();
+		Policy policy = new Policy("/", name, "^foo.", null, 0,
+				Collections.<String, Object>singletonMap("message-ttl", 1000));
+		this.template.addPolicy(policy);
+		List<Policy> list = this.template.policies();
+		assertTrue(list.size() > 0);
+		Policy out = null;
+		for (Policy policyOut : list) {
+			if (policyOut.getName().equals(name)) {
+				out = policyOut;
+				break;
+			}
+		}
+		assertNotNull(out);
+
+		list = this.template.policies("/");
+		assertTrue(list.size() > 0);
+		out = null;
+		for (Policy policyOut : list) {
+			if (policyOut.getName().equals(name)) {
+				out = policyOut;
+				break;
+			}
+		}
+		assertNotNull(out);
+
+		this.template.removePolicy("/", name);
+		list = this.template.policies();
+		out = null;
+		for (Policy policyOut : list) {
+			if (policyOut.getName().equals(name)) {
+				out = policyOut;
+				break;
+			}
+		}
```

Comment:
Hmmm - don't know what happened to that; it was there at one time; must have dropped it during clean up.